Project.toml

```diff
@@ -19,8 +19,15 @@ OrderedCollections = "1"
 SnoopPrecompile = "1"
 julia = "1.6"
 
+[extensions]
+JuMPDimensionalDataExt = "DimensionalData"
+
 [extras]
+DimensionalData = "0703355e-b756-11e9-17c0-8b28908087d0"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
 
 [targets]
-test = ["Test"]
+test = ["DimensionalData", "Test"]
+
+[weakdeps]
+DimensionalData = "0703355e-b756-11e9-17c0-8b28908087d0"
```

Comment:
So what happens with this in Julia 1.6? It just gets ignored?

Reply:
yes

Reply:
the options for Backwards compatibility is by `Requires` or as a regular dependency

Reply:
I think I'm okay if this is a Julia 1.9+ feature only. We don't need backwards compatibility for new features.

One issue is that the docs are currently build using Julia 1.6. So we'd need to update that, and then add some big compat warnings.